app/models/house.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+# represents a dad house
+class House < ApplicationRecord
+  include HousesHelper
+  validates :house_name, :house_start, presence: true, allow_blank: false
+  validate :house_name_unique
+  validate :house_not_old
+  has_many :house_participations, dependent: :nullify
+
+  # @function add_user
+  # @abstract adds the current user
+  def add_user(uid)
+    return can_join?(uid) unless can_join?(uid).blank?
+
+    score = User.find(uid).submissions
+                .where('created_at >= ?', search_date).sum('submissions.time').to_i
+    HouseParticipation
+      .create(user_id: uid, house_id: id, join_date: Time.now.utc.to_date, score: score)
+    ''

```

Comment:
what is this single quote doing here?

Reply:
This is probably my inexperience with Ruby. When there is an error in joining houses, the function returns a string with context for why the user couldn't join a house. When there is no error, the function will ideally return nil/nothing, but because ruby returns the last line of code, it would return the "HouseParticipation" instance that was created. So instead I just had it return an empty string and had it check to see if the string was empty or not to indicate whether or not there was an error. Probably not the best way of doing that I can I can fix that.